### PR TITLE
docs(core): add howto section

### PR DESCRIPTION
**Pull request by vigsterkr**
# Description

adds how-to section to the documentation, describing how-to run renku on HPC with toil and write your own workflow provider

### CODE REVIEW

File: docs/how-to-guides/hpc.rst [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+Running Renku on HPC
+====================
+
+Renku CLI supports various backends for executing workflows. Currently, there
+are two different providers are implemented, namely ``cwltool`` and ``toil``.
+:ref:`provider` documents gives a more detailed description of how to implement
+your own workflow provider.
+
+The default all workflows are executed by the ``cwltool`` provider, that basically
```
Comment (Panaetius) [inline docs/how-to-guides/hpc.rst:11]:
```suggestion
By default all workflows are executed by the ``cwltool`` provider, that
```

File: docs/how-to-guides/hpc.rst [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+exports the workflow to CWL and then uses `cwltool <https://github.com/common-workflow-language/cwltool>`_
+to execute the given CWL.
+
+The workflow backend can be changed by using the ``-p/--provider <PROVIDER>`` and
```
Comment (Panaetius) [inline docs/how-to-guides/hpc.rst:15]:
```suggestion
The workflow backend can be changed by using the ``-p/--provider <PROVIDER>`` 
```

File: docs/how-to-guides/hpc.rst [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+command line option. A backend's default configuration can be overridden by
+providing the  ``-c/--config <config.yaml>`` a command line parameter.
```
Comment (Panaetius) [inline docs/how-to-guides/hpc.rst:17]:
```suggestion
providing the  ``-c/--config <config.yaml>`` command line parameter.
```

File: docs/how-to-guides/hpc.rst [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+The following ``renku`` commands support the above mentioned, workflow provider
```
Comment (Panaetius) [inline docs/how-to-guides/hpc.rst:18]:
```suggestion
The following ``renku`` commands support the above mentioned workflow provider
```

File: docs/how-to-guides/hpc.rst [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+related command line options:
+
+ - :ref:`cli-rerun`,
+ - :ref:`cli-update`,
+ - :ref:`cli-workflow` execute and iterate.
```
Comment (Panaetius) [inline docs/how-to-guides/hpc.rst:23]:
```suggestion
 - :ref:`cli-workflow` ``execute`` and ``iterate``.
```

File: docs/how-to-guides/hpc.rst [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+platforms.
+
+Renku on Slurm
+^^^^^^^^^^^^^^
+`Slurm <https://www.schedmd.com/>`_ is a highly configurable open-source workload manager,
+which is in widespread use at government laboratories, universities and companies world
+wide and performs workload management for over half of the top 10 systems in the TOP500.
+
+As ``toil`` supports Slurm, one can easily execute the previously created renku
+workflows on Slurm, just need to provide a simple configuration file to the provider
```
Comment (Panaetius) [inline docs/how-to-guides/hpc.rst:44]:
```suggestion
workflows on Slurm. One just needs to provide a simple configuration file to the provider
```

File: docs/how-to-guides/index.rst [hunk 7]
```diff
@@ -0,0 +1,15 @@
+.. _how_to_guides:
+
+How-to guides
```
Comment (Panaetius) [inline docs/how-to-guides/index.rst:3]: @rokroskar Do we need to include this in the main docs somehow?

File: docs/how-to-guides/provider.rst [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+        from typing import Any, Dict, List
+
+        from renku.core.models.workflow.provider import IWorkflowProvider
+        from renku.core.plugins import hookimpl
+
+        class MyProvider(IWorkflowProvider):
+            @hookimpl
+            def workflow_provider(self):
+                """Workflow provider name."""
+                return (self, "my")
```
Comment (Panaetius) [inline docs/how-to-guides/provider.rst:32]: We should mention that this is supposed to return a unique identifier string for this plugin

File: docs/how-to-guides/provider.rst [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+                # traversing the dag that contains the ``Plans``
+                # and executing each plan in the graph
+                ...
+
+                return generated_outputs
+
+The execution of the workflow(s) shall be defined in ``workflow_execute`` function, where
+
+  - ``dag`` is a Directed Acyclic Graph of ``Plans`` to be executed,
```
Comment (Panaetius) [inline docs/how-to-guides/provider.rst:47]: Mention networkx ?
And we could link to the https://renku.readthedocs.io/projects/renku-python/en/latest/reference/models/workflow.html#renku.core.models.workflow.plan.Plan section

**Review by Panaetius — CHANGES_REQUESTED** [review]
Thanks! I think this will be very helpful to users

File: docs/how-to-guides/index.rst (hunk 7 above)
Comment (rokroskar) [reply docs/how-to-guides/index.rst:3]: the idea was to create a link there to this section. renku-python is now a submodule of the main renku repo for this purpose. Please make an issue for it in the renku repo for now.

Comment (vigsterkr) [reply docs/how-to-guides/index.rst:3]: https://github.com/SwissDataScienceCenter/renku/issues/2379

**Review by Panaetius — APPROVED** [review]

**Pull request merged**